test/e2e/flask/petnames-transactions.spec.js

```diff
@@ -33,15 +45,66 @@ describe('Petnames', function () {
       async ({ driver }) => {
         await unlockWallet(driver);
         await openDapp(driver);
-        await createTransactionRequest(driver);
+        await createTransactionRequestInDapp(driver);
         await switchToNotificationWindow(driver, 3);
         await expectName(driver, '0x0c54F...7AaFb', false);
-        await saveName(driver, '0x0c54F...7AaFb', 'test.lens', undefined);
+
+        // Test custom name.
+        await saveName(driver, '0x0c54F...7AaFb', 'Custom Name', undefined);
         await rejectSignatureOrTransactionRequest(driver);
         await focusTestDapp(driver);
-        await createTransactionRequest(driver);
+        await createTransactionRequestInDapp(driver);
         await switchToNotificationWindow(driver, 3);
-        await expectName(driver, 'test.lens', true);
+        await expectName(driver, 'Custom Name', true);
+
+        // Test proposed name.
+        await saveName(driver, 'Custom Name', undefined, 'test4.lens', true);
+        await rejectSignatureOrTransactionRequest(driver);
+        await focusTestDapp(driver);
+        await createTransactionRequestInDapp(driver);
+        await switchToNotificationWindow(driver, 3);
+        await expectName(driver, 'test4.lens', true);
+      },
+    );
+  });
+
+  it('can update custom names for addresses in transaction requests', async function () {
```

Comment:
As the mean difference between these tests is the type of transaction, should we instead name them something like `can save petnames for addresses in dapp transactions` and `can save petnames for addresses in send transactions`?

Reply:
See note above about naming. I've gone with dapp vs wallet. 

Let me know your thoughts.